### PR TITLE
Add focus assertions

### PR DESCRIPTION
**Pull request by EsteveAguilera**
This PR includes `assertHasFocus` and `assertHasNotFocus` as requested on the issue #95

The EditTexts used to test the two new assertions have been added to the SomeViewsWithDifferentVisibilitesActivity.

I am not sure if it's the best place as it talks about visibility, but I thought it's not worth to add a new Activity just to add this two tests.

### CODE REVIEW

**Comment by EsteveAguilera** [comment]
Ooops, the conflicts are from the PR  #149 that renames a class...

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java [hunk 1]
```diff
@@ -279,4 +281,14 @@ public void checkSameDrawable() throws Exception {
...
     assertDrawable(R.id.image_view, R.drawable.ic_action_menu);
   }
+
+  @Test
+  public void checkViewHasFocus() throws Exception {
+    assertHasFocus(R.id.edittext_with_focus);
+  }
+
+  @Test
+  public void checkViewHasNotFocus() throws Exception {
```
Comment (rocboronat) [inline sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java]: Add these new methods to `methodsByIdLaunchExceptionWhenThePassedIdIsNotAResIdOrAStringId()` test, to check that they give a great error message if the passed int is not an id.

Comment (EsteveAguilera) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java]: Oh, ok!

Comment (Sloy) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java]: In this case the method doesn't support string res, only id. 
If we don't do any magic, we won't throw any special error like the other cases in that test.

In this case I think Espresso's message will be good enough.

Comment (EsteveAguilera) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java]: You're right @Sloy , but I think I can do this PR better by allowing to access the view not only by `id`, but with a text, e.g for a button!

Give me some time and will add that

File: library/src/main/java/com/schibsted/spain/barista/assertion/BaristaAssertions.kt [hunk 2]
```diff
@@ -181,4 +181,14 @@ object BaristaAssertions {
...
         onView(withId(id)).check(matches(withDrawable(drawable)))
     }
+
+    @JvmStatic
+    fun assertHasFocus(@IdRes id: Int) {
+        onView(withId(id)).check(matches(hasFocus()))
+    }
+
+    @JvmStatic
+    fun assertHasNotFocus(@IdRes id: Int) {
```
[the remark this answers is not in the document]
Comment (EsteveAguilera) [reply library/src/main/java/com/schibsted/spain/barista/assertion/BaristaAssertions.kt]: I'm quite bad at naming too, and I also thought about it for a while

After seeing the code `(not(hasFocus()))` and the issue title `Assert view has / not has focus`, I decided to leave the method name as is

I agree that `assertIsFocused`and `assertIsNotFocused` maybe sounds better... Let me call @rocboronat to see what he thinks about it!

[the remark this answers is not in the document]
Comment (rocboronat) [reply library/src/main/java/com/schibsted/spain/barista/assertion/BaristaAssertions.kt]: Looking at the style of the Barista's API, we must call it `assertIsFocused` and `assertNoFocused` to match the other methods style. If not, we should rename `assertEnabled` to `assertIsEnabled` as well as the other methods 😢

[the remark this answers is not in the document]
Comment (EsteveAguilera) [reply library/src/main/java/com/schibsted/spain/barista/assertion/BaristaAssertions.kt]: 👍

[the remark this answers is not in the document]
Comment (Sloy) [reply library/src/main/java/com/schibsted/spain/barista/assertion/BaristaAssertions.kt]: Makes sense, I like that format too. 🎉

**Pull request merged**